### DISCUSSION TITLE
Partial property for AOT compatibility and MVVMTK0045

### DISCUSSION THREAD

**Discussion by King-Taz** [CommunityToolkit/dotnet#1035] [General]
Using the code below

```
    [ObservableProperty]
    public int fontSize

    // This method called whenever `fontSize` changes.
    partial void OnFontSizeChanged(int value)
    {
        Preferences.Set("FontSize", value);
    }
```

I get the MVVMTK0045 error

```
The field *.*.fontSize using [ObservableProperty] will generate code that is not AOT compatible...
and a partial property should be used instead
```

So I change the codes as suggested in the documentation to 

```
    [ObservableProperty]
    public partial int FontSize { get; set }

    // This method called whenever `fontSize` changes.
    partial void OnFontSizeChanged(int value)
    {
        Preferences.Set("FontSize", value);
    }
```

does not work and creates a different error "the partial property must have an implementation part". Any idea how to correct this issue?

**Comment by FifthCloud** [comment]
I wish I knew why this worked, but I followed your exact steps. However, the only thing I added was within the csproj was add...

`<LangVersion>preview</LangVersion>`

The errors went away. (shrug)
